src/propagators/text_map_codec.js

```diff
@@ -49,8 +49,14 @@ export default class TextMapCodec {
     }
 
     _decodedValue(value: string): string {
-        if (this._urlEncoding) {
-            return decodeURIComponent(value);
+        // only use url-decoding if there are meta-characters '%'
+        if (this._urlEncoding && value.indexOf('%') > -1) {
+            // unfortunately, decodeURIComponent() can throw 'URIError: URI malformed'
+            try {
+                return decodeURIComponent(value);
```

Comment:
@saminzadeh do you know if there's an alternative std function that never throws exceptions?